### DISCUSSION TITLE
Any plans to support collision between MeshShape？

### DISCUSSION THREAD

**Discussion by feixinyv** [jrouwe/JoltPhysics#202] [Q&A] [answered]
Hello, this project is awesome. 
I found that MeshShape can only be static, or we can have only one dynamic MeshShape in the whole scene. 
Do you have any plans to support multiple dynamic MeshShapes which can colliding with each other？
Does Horizon Forbidden West lives without collision between dynamic MeshShapes？

**Comment by jrouwe** [comment, marked as answer]
Hello,
I started implementing dynamic meshes but I'm still on the fence if they should actually be supported. The reasons being:

1. Mesh vs mesh collision is incredibly expensive, most games don't have the CPU cycles to spare for this.
2. With mesh vs mesh collision the only thing you can do is push triangles apart (because there's no concept of solid volume), but depending on how your meshes are constructed this may not actually separate the two meshes so it's easy for things to get stuck, especially if they start with a lot of interpenetration.

I never used dynamic meshes. Normally you would construct your dynamic objects as either a convex hull or as a compound of convex hulls. You can build those manually or you could use convex decomposition for this.

**Comment by DocAce** [comment]
For automatically decomposing meshes into a compound of convex hulls I can recommend https://github.com/kmammou/v-hacd

**Reply by feixinyv** [reply]
Thanks for the information!
